Sources/SwiftGenKit/Parsers/Files/FilesFile.swift

```diff
@@ -0,0 +1,42 @@
+//
+// SwiftGenKit
+// Copyright © 2020 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+
+extension Files {
+  struct File {
+    let name: String
+    let ext: String?
+    let path: [String]
+    let mimeType: String
+
+    init(path: Path, relativeTo parent: Path? = nil) throws {
+      guard path.exists else {
+        throw ParserError.invalidFile(path: path, reason: "Unable to read file")
+      }
+
+      self.ext = path.extension
+      self.name = path.lastComponentWithoutExtension
+      if let relative = parent.flatMap({ path.relative(to: $0) })?.parent(),
```

Comment:
Can this bit be simplified to this?
```swift
let relativeParent = (parent.flatMap({ path.relative(to: $0) }) ?? path).parent()
self.path = relativeParent.components.drop { $0 == "." }
```

Also not 100% sure if that `.parent()` is needed?

Reply:
Hmm. Then if `parent` was null, you could have an absolute path, no? `self.path` needs to be relative to some root directory, and shouldn't contain the filename

Reply:
Honestly dunno, it was late last night and just quickly wrote something 😅 Ignore the `.parent()` question.

I haven't checked out this branch yet, and played around with it. Maybe this? 
```swift
let relativeParent = parent.flatMap({ path.relative(to: $0) })?.parent() ?? ""
self.path = relativeParent.components.drop { $0 == "." }
```

Reply:
Put in something similar to this. Thanks!